### PR TITLE
Feat(plugins): Add ISIS support to encrypt and decrypt filters

### LINKED ISSUE #3624: Add ISIS support in arista.avd.encrypt plugin
### Enhancement summary

Currently the plugin supports encrypting BGP and OSPF clear text passwords into EOS passwords.

We have a project with ISIS, and we would need the same functionality for ISIS.

### Which component of AVD is impacted

plugins

### Use case example

SP project using ISIS as routing-protocol and AAP to store credentials in clear text.

### Describe the solution you would like

Extend the plugin to support ISIS passwords.

https://avd.arista.com/4.5/roles/eos_cli_config_gen/docs/input-variables.html#router-isis

We can set already ISIS authentication keys using eos_cli_config_gen:
```
    isis_authentication_mode: <str; "text" | "md5">

    # Type-7 encrypted password
    isis_authentication_key: <str>
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add ISIS support to encrypt and decrypt filters

## Related Issue(s)

Fixes #3624 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Add `passwd_type: isis` option to `arista.avd.encrypt` and `arista.avd.decrypt` filters.
- Requires `key: <isis_instance_name>` and `mode: <none | text | md5 | sha | sha-1 | sha-224 | sha-256 | sha-384 | sha1-512>`

## How to test

- [x] Added unit tests
- [x] Added integration tests

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py [hunk 1]
```diff
@@ -160,3 +160,79 @@ def bgp_decrypt(password: str, key) -> str:
...
+    "sha": "sha",
+    "sha-1": "sha_1",
+    "sha-224": "sha_224",
+    "sha-256": "sha_256",
+    "sha-384": "sha_384",
+    "sha-512": "sha_512",
+}
+
+
+def _validate_isis_args(password: str, key: str, mode: str):
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:181]: could we use Literal for `mode`?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:181]: I think we should do that for all filter functions once we move them to pyavd. currently they are only used by filters, where type hinting won't help.

**Review by gmuloc — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py [hunk 2]
```diff
@@ -160,3 +160,79 @@ def bgp_decrypt(password: str, key) -> str:
...
+        raise AristaAvdError("Password is required for encryption/decryption")
+
+    if not isinstance(password, str):
+        raise AristaAvdError(f"Password MUST be of type 'str' but is of type {type(password)}")
+
+    if not isinstance(key, str):
+        raise AristaAvdError(f"Key MUST be of type 'str' but is of type {type(key)}")
+
+    if not isinstance(mode, str):
+        raise AristaAvdError(f"Mode MUST be a string with one of the following options: {list(_ISIS_MODE_MAP)}. Got '{mode}'.")
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:192]:
```
for arg, arg_type in zip([password, key, mode], [str, str, str]):
  if not isinstance(arg, arg_type):
    raise AristaAvdError(f"{arg} MUST be of type '{arg_type.__name__}' but is of type {type(arg).__name__}")
```

File: ansible_collections/arista/avd/tests/unit/plugins/filter/test_password.py [hunk 3]
```diff
@@ -181,6 +183,62 @@ def test_ospf_message_digest_decrypt_failure(key, password, hash_algorithm, key_
...
+
+ISIS_INVALID_INPUT_DECRYPT = [
+    pytest.param("test", "none", "3QGcqpU2YTwKh2jVQ4Vj/A==", id="Wrong password"),
+]
+
+
+@pytest.mark.parametrize("key, mode, password, encrypted_password", ISIS_INPUT_EXPECTED)
+def test_isis_encrypt(key, mode, password, encrypted_password):
+    """
+    Test isis_simple_encrypt
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/tests/unit/plugins/filter/test_password.py]:
```suggestion
    Test isis_encrypt
```

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:192]: While I understand your suggestion and DRY etc. I will not implement this right now.

File: ansible_collections/arista/avd/tests/unit/plugins/filter/test_password.py (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/tests/unit/plugins/filter/test_password.py]: Too much copy paste :) Thanks!

**Comment by carlbuchmann** [comment]
@dgonzalez85 - please test and review this PR!

**Review by MaheshGSLAB — APPROVED** [review]

**Comment by dgonzalez-arista** [comment]
Hi, i have done a bit of testing around this, enabling authentication between a couple of links:

1) note in eos_cli_config_gen we support today text and md5 authentication only according to docs. The plugin function supports addtional hashing methods:
https://avd.arista.com/4.6/roles/eos_cli_config_gen/docs/input-variables.html#ethernet-interfaces

2) Just a note regarding probably old comments in this issue, decrypt plugin parameter is "mode" and no "isis_mode"

3) I used the following configuration to test md5 auth and works great

```
---
custom_structured_configuration_ethernet_interfaces:
  - name: "Ethernet1"
    isis_authentication_mode: "md5"
    isis_authentication_key: "{{ isis_peerings.password | arista.avd.encrypt(passwd_type='isis', mode='md5', key='EVPN_UNDERLAY') }}"
```

Which worked fine i could generate the configurations and session is up and MD5 auth enabled:
```
DC1-CL1#show run int et1
interface Ethernet1
   description P2P_LINK_TO_DC1-SP1_Ethernet1
   no shutdown
   mtu 1500
   speed forced 100gfull
   no switchport
   ip address 192.168.1.1/31
   isis enable EVPN_UNDERLAY
   isis circuit-type level-2
   isis metric 50
   isis network point-to-point
   isis authentication mode md5
   isis authentication key 7 $1c$b5bHgaRyiaJARvaJb6EneQ==
   
DC1-CL1#show isis int et1

IS-IS Instance: EVPN_UNDERLAY VRF: default

  Interface Ethernet1:
    Index: 48 SNPA: P2P
    MTU: 1497 Type: point-to-point
    Supported address families: IPv4
    Area proxy boundary is disabled
    Speed: 100000 mbps
    BFD IPv4 is disabled
    BFD IPv6 is disabled
    Hello padding is enabled
    Level 2:
      Metric: 50, Number of adjacencies: 1
      Link-ID: 30
      Authentication mode: MD5
      TI-LFA protection is disabled for IPv4
      TI-LFA protection is disabled for IPv6
	  
DC1-CL1#show isis nei
 
Instance  VRF      System Id        Type Interface          SNPA              State Hold time   Circuit Id          
EVPN_UNDE default  DC1-SP1          L2   Ethernet1          P2P               UP    25          34                  
EVPN_UNDE default  DC1-SP2          L2   Ethernet2          P2P               UP    24          2C                  
EVPN_UNDE default  DC1-CL2          L2   Vlan4093           P2P               UP    24          24                  
```
4) I used the following configuration to test text auth and also all good:

```
---
custom_structured_configuration_ethernet_interfaces:
  - name: "Ethernet1"
    isis_authentication_mode: "text"
    isis_authentication_key: "{{ isis_peerings.password | arista.avd.encrypt(passwd_type='isis', mode='text', key='EVPN_UNDERLAY') }}"

DC1-SP1#show isis int et1

IS-IS Instance: EVPN_UNDERLAY VRF: default

  Interface Ethernet1:
    Index: 52 SNPA: P2P
    MTU: 1497 Type: point-to-point
    Supported address families: IPv4
    Area proxy boundary is disabled
    Speed: 100000 mbps
    BFD IPv4 is disabled
    BFD IPv6 is disabled
    Hello padding is enabled
    Level 2:
      Metric: 50, Number of adjacencies: 1
      Link-ID: 34
      Authentication mode: Text
      TI-LFA protection is disabled for IPv4
      TI-LFA protection is disabled for IPv6

DC1-SP1#show run int et1
interface Ethernet1
   description P2P_LINK_TO_DC1-CL1_Ethernet1
   no shutdown
   mtu 1500
   speed forced 100gfull
   no switchport
   ip address 192.168.1.0/31
   isis enable EVPN_UNDERLAY
   isis circuit-type level-2
   isis metric 50
   isis network point-to-point
   isis authentication mode text
   isis authentication key 7 2ewYXSX6oRxJvMW6Qzn1fA==
```

**Review by dgonzalez-arista — APPROVED** [review]
Tested md5 and text authentication and looking good!

**Pull request merged**